src/Monitoring/Sdk/GrafanaClient.cs

```diff
@@ -0,0 +1,246 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Monitoring.Sdk
+{
+    public sealed class GrafanaClient : IDisposable
+    {
+        private readonly HttpClient _client;
+
+        // e.g. https://dotnet-eng-grafana.westus2.cloudapp.azure.com/
+        private readonly string _baseUrl;
+
+        public GrafanaClient(string baseUrl, string apiToken)
+        {
+            _baseUrl = baseUrl;
+            _client = new HttpClient();
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiToken);
+        }
+
+        public async Task<JObject> GetDashboardAsync(string uid)
+        {
+            var uri = new Uri(new Uri(_baseUrl), $"/api/dashboards/uid/{uid}");
+            using (HttpResponseMessage response = await _client.GetAsync(uri).ConfigureAwait(false))
+            {
+                response.EnsureSuccessStatusCode();
+
+                using (Stream stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                using (var streamReader = new StreamReader(stream))
+                using (var jsonReader = new JsonTextReader(streamReader))
+                {
+                    return await JObject.LoadAsync(jsonReader).ConfigureAwait(false);
+                }
+            }
+        }
+
+        public async Task<JArray> ListFoldersAsync()
+        {
+            var uri = new Uri(new Uri(_baseUrl), "/api/folders?limit=50");
```

Comment:
Why limit? I mean, we'll very probably not care, but curious if the API required it?

Reply:
I'm... not sure what will happen if you don't pass the limit.  Didn't know if it was going to default pick some annoyingly low number.

I could raise it (I think it said 5000 was the maximum valid limit), but I just wanted to be sure we didn't run into the undocumented default being unuseful.